DetectorDescription/DDCMS/interface/DDSolidShapes.h

```diff
@@ -77,6 +99,31 @@ namespace cms {
        {DDSolidShape::ddellipticaltube, "EllipticalTube"},
        {DDSolidShape::ddcuttubs, "CutTube"},
        {DDSolidShape::ddextrudedpolygon, "ExtrudedPolygon"}}};
+
+  const std::array<const cms::dd::ValuePair<LegacySolidShape, cms::DDSolidShape>, 22> LegacySolidShapeMap{
+      {{LegacySolidShape::dd_not_init, cms::DDSolidShape::dd_not_init},
+       {LegacySolidShape::ddbox, cms::DDSolidShape::ddbox},
+       {LegacySolidShape::ddtubs, cms::DDSolidShape::ddtubs},
+       {LegacySolidShape::ddtrap, cms::DDSolidShape::ddtrap},
+       {LegacySolidShape::ddcons, cms::DDSolidShape::ddcons},
+       {LegacySolidShape::ddpolycone_rz, cms::DDSolidShape::ddpolycone},
+       {LegacySolidShape::ddpolycone_rz, cms::DDSolidShape::ddpolycone},
+       {LegacySolidShape::ddpolycone_rrz, cms::DDSolidShape::ddpolycone},
+       {LegacySolidShape::ddpolyhedra_rz, cms::DDSolidShape::ddpolyhedra},
+       {LegacySolidShape::ddpolyhedra_rz, cms::DDSolidShape::ddpolyhedra},
```

Comment:
Why are there duplicate lines? If they are not a mistake, they need a comment explaining them so I don't delete them some time in the future. What is the significance of the number 22?

Reply:
@cvuosalo - ```std::array``` is a container that encapsulates fixed size arrays. If you change this number by mistake the compiler will complain :-)